### ISSUE TITLE
importScripts 又怎么引用呢

### ISSUE THREAD

**Issue by moweng** [darionco/rollup-plugin-web-worker-loader#1]

**Comment by darionco** [comment]
Google translates to:
```
importScripts how to quote it?
```
I'm not sure what you are asking, could you explain with more detail?

**Comment by PixlRainbow** [comment]
I think they are asking how to bundle the dependencies of the Web Worker itself.
In vanilla JS, you import external scripts inside of a web worker by stating
```js
importScripts('path/to/script1', 'path/to/script2', ...)
```
However, most bundlers I have seen seem to lack the ability to scan the worker script itself for dependencies, or require the worker to be written in node.js/ES6 module style.

**Comment by darionco** [comment]
Yeah, this one is the same, dependencies must be imported using node.js/ES6 module style.

If there is enough traction for this I could make it into a feature. I'll leave open this issue for a while and see if anyone else needs to be able to use `importScripts` rather than ES6 `import` statements.

**Comment by PixlRainbow** [comment]
Personally, my workaround is to write an if else check for whether environment variables injected by bundling tools are defined
```js
if(typeof PRODUCTION !== 'undefined'){
    var name = require("name");
}
else{
    importScripts("node_modules/name");
}
```

**Comment by PixlRainbow** [comment]
## UPDATE
Although that trick works with webpack, I have run into a slight problem:  
rollup needs rollup-plugin-commonjs in order to parse `require` statements. There appears to be a bug when that plugin interacts with your plugin; the commonjs plugin will insert `export default` into the end of the worker blob, which makes the worker code invalid. From what I can tell, the commonjs plugin assumes that the blob produced from your plugin should become an ES6 module instead of iife

**Comment by darionco** [comment]
that's not good, any chance you could put together a small project that reproduces the issue?

Thanks!

**Comment by PixlRainbow** [comment]
[badroll.zip](https://github.com/darionco/rollup-plugin-web-worker-loader/files/3841486/badroll.zip)
Here you go! Note that I have accidentally picked a library (file-saver) that doesn't actually do anything when run inside a worker, but normally would not throw any exceptions either. When `bundle.js` is created and `main.html` is loaded in the browser, there is immediately an exception thrown as the worker script is parsed:  
![image](https://user-images.githubusercontent.com/44422178/68770820-d5fddd00-0661-11ea-8905-ad569914cf66.png)  
The highlighted line in the exception is `export default worker`

**EDIT**
Originally, I did not use `import` statement as they could not be wrapped inside `if` statements (rollup only allows `import` statements at top-level, not inside any scope). However, I later found [`rollup-plugin-jscc`](https://www.npmjs.com/package/rollup-plugin-jscc) which provides conditional compilation functionality. Pairing that with `import`, I finally got to create JavaScript that was both directly runnable in-browser and bundleable. Rollup's native built-in `import` bundler appears to handle your worker blobs just fine.
[goodroll.zip](https://github.com/darionco/rollup-plugin-web-worker-loader/files/3842064/goodroll.zip)

**EDIT 2**
I have just tried pairing this plugin with [`rollup-plugin-babel`](https://www.npmjs.com/package/rollup-plugin-babel) and found that for some reason, the web worker runs *after* babel and doesn't get its helper functions (e.g. `createBase64WorkerFactory`) transpiled to ES5 even though I ordered the babel plugin after your plugin.

**Comment by darionco** [comment]
I am looking into fixing the incompatibility with `rollup-plugin-commonjs` and opened a new issue #6 to track this `rollup-plugin-babel` issue. If possible, could you comment there and add a project to repro the issue?

Thanks so much for your help!

**Comment by darionco** [comment]
Fixed in version 0.7.0
